File: leetcode/python/0121_best_time_to_buy_and_sell_stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit, minPrices = 0, 1e8
        # loop through prices using i:
        # -> if minPrices > prices[i]: minPrices = prices[i]
        # -> profit = max(profit, prices[i] - minPrices)
        # return profit
        for i in range(len(prices)):
            if minPrices > prices[i]: minPrices = prices[i]
            profit = max(profit, prices[i] - minPrices)
        return profit

File: leetcode/java/0121_best_time_to_buy_and_sell_stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrices = Integer.MAX_VALUE, profit = 0;

        for (int i = 0; i < prices.length; i++) {
            if (minPrices > prices[i]) minPrices = prices[i];
            if (profit < prices[i] - minPrices) profit = prices[i] - minPrices;
        }

        return profit;
    }
}